pyflakes/test/test_api.py

```diff
@@ -372,8 +375,8 @@ def test_missingTrailingNewline(self):
         exception to be raised nor an error indicator to be returned by
         L{check}.
         """
-        fName = self.makeTempFile("def foo():\n\tpass\n\t")
-        self.assertHasErrors(fName, [])
+        with self.makeTempFile("def foo():\n\tpass\n\t") as fName:
```

Comment:
if you dont mind, when rebasing, could you also change the double quotes to single quotes on lines that you modify, where that is sensible.

It isnt a firm style policy of this project, but seems to be were the codebase is going.

or if you prefer, I can do the rebasing for you.

Reply:
If we just switch to `black` then we can stop nit-picking this kind of nonsense.

Reply:
or if you really want single quotes, you could use [`double-quote-string-fixer`](https://github.com/pre-commit/pre-commit-hooks)

Reply:
imo, unless you have an automated tool enforcing it (such as [pre-commit](https://pre-commit.com)) it's not worth bringing up or rebasing commits -- just me though :man_shrugging:

Reply:
Firstly, please note rebasing is about a linear git history.  That was instituted for this project long before I became a regular contributor and maintainer, but I do appreciate the value of it vs the GitHub flow approaches.

Other than that, I agee. I am also of the opinion that automated tools enforcing rules is helpful, but not at the expense of massive rewrites for the sake of it.  I also run @coala , and that framework with the https://gitmate.io integration has the ability to lint only the modified lines according to repository rules, and report back on PRs to helpfully guide newcomers, including advising that a rebase is required, and providing a GitHub based command to do the rebase automatically, so then at least it is clear that the committer was an automation and didnt perform any additional changes to the author's commit.

It would be great to have clearer contributing guidelines and a properly enforced code of conduct, to make this project a bit more enjoyable to contribute in, and ideally add advanced tools to help, but by consensus of the existing contributors.  We have been a bit lax in that department.

Getting back to the rebasing thing, are you ok with having someone else as the committer?
It can be a bit offputting on GitHub as it show multiple avatars on the commit.  I like to be sure before I do that to someone elses PR.

Reply:
I assume you'd just click `rebase and squash` when merging if you're trying to enforce a linear history -- this already causes "Authored by ... Committed by ..." so....

Reply:
fwiw, I would rebase, and repush, to ensure the changes pass on top of the current master.  Sometimes it is easy to know that this is unnecessary, but it is better to do it systematically.  That was what the maintainers were before I was a committer, and I have continued to do.

And it is what many other orgs insist on, especially any on GitLab which makes that workflow seemless.

There are many other benefits of it, which are not so noticable inside GitHub.  But for example, the Squash and Merge creates a new commit, even for commits which are already on top of master.

Anyways, seems you are ok with it, so I'll proceed.

Reply:
indeed, go for it!  personally I don't like rebase-squash because I can never know if my actual patch was merged or not but if that's how the project runs do your worst :)

Reply:
also in this case you didn't even need to rebase my commit, since it gets squashed at merge

Reply:
Ya, that is the downside, which is why @coala requires that the contributor does it themselves, or they invoke the bot to do it, but this project hasnt modernised.  But with more regular contributors, maybe we can change that. ;-)

Reply:
alternatively you can just re-trigger the `/pr` CI jobs -- travis / appveyor will use the latest "merge" and then you don't need everyone to rebase every time you merge a PR